dotcom-rendering/src/web/components/LiveEpic.importable.tsx

```diff
@@ -0,0 +1,288 @@
+			hasOptedOutOfArticleCount: !consentGivenForArticleCounts(consent),
+			modulesVersion: 'v3',
+			url: window.location.origin + window.location.pathname,
+		},
+	};
+};
+
+/**
+ * Render
+ *
+ * Dynamically imports and renders a given module
+ *
+ * @param url string - The url of the compoenent
+ * @param name string - The name of the compoenent
+ * @param props object - The props of the compoenent
+ * @returns The resulting react component
+ */
+const Render = ({
+	url,
+	name,
+	props,
+}: {
+	url: string;
+	name: string;
+	props: { [key: string]: unknown };
+}) => {
+	// Using state here to store the Epic component that gets imported allows
+	// us to render it with React (instead of inserting it into the dom manually)
+	const [Epic, setEpic] = useState<React.FC<{ [key: string]: unknown }>>();
+
+	useEffect(() => {
+		setAutomat();
+		window
+			.guardianPolyfilledImport(url)
+			.then((epicModule) => {
+				// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+				setEpic(() => epicModule[name]); // useState requires functions to be wrapped
+			})
+			.catch((err) => {
+				const msg = `Error importing LiveBlog epic: ${err}`;
+				window.guardian.modules.sentry.reportError(
+					new Error(msg),
+					'liveblog-epic',
+				);
+			});
+	}, [url, name]);
+
+	if (!Epic) return null;
+	log('dotcom', 'LiveEpic has the Epic');
+
+	return (
+		<aside
+			css={css`
+				margin-bottom: ${space[3]}px;
+			`}
+		>
+			{/* eslint-disable-next-line react/jsx-props-no-spreading */}
+			<Epic {...props} />
+		</aside>
+	);
+};
+
+const Fetch = ({
+	payload,
+	contributionsServiceUrl,
+}: {
+	payload: unknown;
+	contributionsServiceUrl: string;
+}) => {
+	let url = joinUrl(contributionsServiceUrl, 'liveblog-epic');
+
+	// Check if we've been asked to force this epic
+	const params = new URLSearchParams(window.location.search);
+	if (params.get('force-liveblog-epic')) {
+		url += '?force=true';
+	}
+
+	// Send the payload to Contributions to request a module
+	const { data } = useApi<{
+		data: {
+			module: {
+				url: string;
+				name: string;
+				props: { [key: string]: unknown };
+			};
+		};
+	}>(
+		url,
+		{
+			revalidateIfStale: false,
+			revalidateOnFocus: false,
+			revalidateOnReconnect: false,
+		},
+		{
+			method: 'POST',
+			headers: {
+				Accept: 'application/json',
+				'Content-Type': 'application/json',
+			},
+			body: JSON.stringify(payload),
+		},
+	);
+
+	// If we didn't get a module in response (or we're still waiting) do nothing. If
+	// no banner should be shown data is equal to {}, hence the Object.keys
+	if (!data || Object.keys(data).length === 0) return null;
+	log('dotcom', 'LiveEpic has a module');
+
+	// Take any returned module and render it
+	return (
+		<Render
+			url={data.data.module.url}
+			name={data.data.module.name}
+			props={data.data.module.props}
+		/>
+	);
+};
+
+/**
+ * LiveEpic is the Support the Guardian banner which appears in-between blocks in blogs
```

Comment:
Can we call it `LiveblogEpic` to be extra clear about where it appears?
Also, it's best not to say "banner" in this context as the RR epic and banner are very distinct things!

Reply:
> Can we call it `LiveblogEpic` to be extra clear about where it appears? 

Good idea, I'll make this change now.

> Also, it's best not to say "banner" in this context as the RR epic and banner are very distinct things!

Thanks. It's actually useful to have this explained, I'll make sure to check any comments so they follow this pattern

Reply:
Thanks! Additional context -
In RR "banner" always means the popup at the bottom (`StickyBottomBanner`), which is generally not content-aware.
The epic is inserted into the content (either at the end of an article, or as a liveblog block), and is often related to the content.